payroll-processor-client/src/app/employee/employee-create/employee-create.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { throwError } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+
+import { Employee } from '../employee-list/state/employee-list.model';
+import { EmployeeListStore } from '../employee-list/state/employee-list.store';
+
+import { EnvService } from 'src/app/shared/env.service';
+
+@Injectable({ providedIn: 'root' })
+export class EmployeeCreateService {
+  private readonly apiUrl: string;
+
+  constructor(
+    private http: HttpClient,
+    private employeeListStore: EmployeeListStore,
+    // private router: Router,
+    envService: EnvService,
+  ) {
+    this.apiUrl = envService.apiRootUrl;
+  }
+
+  createEmployee(employee: Employee) {
+    console.log(employee);
+    return this.http
+      .post<Employee>(`${this.apiUrl}/employees`, employee)
+      .pipe(
+        tap((detail) => {
+          console.log(detail);
+          this.employeeListStore.upsert(detail.id, {
```

Comment:
I believe we can pass the employee directly in as the 2nd parameter here:

`this.employeeListStore.upsert(detail.id, detail)`

However, the API doesn't currently return the created employee, instead the list will need to re-query for it.

How do you want to handle that? Should the API return the created entity (it can!) or should we keep everything asynchronous and require the app poll for updates?

(This would be a great place to add SignalR to push an update to the app)

Reply:
So after discusson, the flow for now will be to return the detail in a restful manner as we would expect from an api. In the future, the functions will queue up pending employees with a null response causing the client to insert the employee into employee-list store locally and as they are created in functionland, signalR will push updates back to the with the "real" persisted data to display to the client.